cinema/views.py

```diff
@@ -1,45 +1,106 @@
-from rest_framework.decorators import api_view
-from rest_framework.response import Response
-from rest_framework import status
-
 from django.shortcuts import get_object_or_404
-
-from cinema.models import Movie
-from cinema.serializers import MovieSerializer
+from rest_framework import generics, mixins, viewsets, status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from cinema.models import Movie, CinemaHall, Actor, Genre
+from cinema.serializers import (MovieSerializer,
+                                CinemaHallSerializer,
+                                ActorSerializer,
+                                GenreSerializer)
 
 
-@api_view(["GET", "POST"])
-def movie_list(request):
-    if request.method == "GET":
-        movies = Movie.objects.all()
-        serializer = MovieSerializer(movies, many=True)
+class GenreList(APIView):
```

Comment:
Consider using `generics.ListCreateAPIView` for `GenreList` to simplify the code and reduce boilerplate.